x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -233,11 +239,12 @@ export class AlertsClient {
     const id = options?.id || SavedObjectsUtils.generateId();
 
     try {
-      await this.authorization.ensureAuthorized(
-        data.alertTypeId,
-        data.consumer,
-        WriteOperations.Create
-      );
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: data.alertTypeId,
```

Comment:
Annoying naming comment coming up :) Doesn't have to be addressed in this PR, but should we revisit naming? There's `alertTypeId` which is the registered alerting type, and the `alertingType` that's either `rule` or `alert` and then `alertTypeId` is also known as `ruleTypeId`, right?

Reply:
Unfortunately, we can't get rid of `alertTypeId` altogether because that is how it is stored in the rule saved object.

I tried to update the AuthorizationClient to reflect the new terminology, that is why `data.alertTypeId` is passed in as `ruleTypeId`. I will likely try to minimize other changes to the alerts client in this PR.

Open to suggestions for what to call the `rule/alert` enum. I went to `AlertingAuthorizationType` but definitely not set on that.

Reply:
Renamed type to entity https://github.com/elastic/kibana/pull/99078/commits/bfbfae406722dc86bbfa235c16e618228c3f6ebe